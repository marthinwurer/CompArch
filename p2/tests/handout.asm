;
; sample program from project handout
;
	.globl	start
base = .
. = base + 76
;
start:
	clr	r3
	mov	r2,(r4)
	bpt
;
	mov	x,r2
	bpt
;
	mov	#123,r3
	bpt
;
	halt
;
. = base + 214
;
x:	.word	-1
;
	.end	start
