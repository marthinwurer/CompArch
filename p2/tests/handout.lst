      1                                 ;
      2                                 ; sample program from project handout
      3                                 ;
      4 000000                                   .globl  start
      5 000000                          base = .
      6 000076                          . = base + 76
      7                                 ;
      8 000076                          start:
      9 000076  005003                           clr     r3
     10 000100  010214                           mov     r2,(r4)
     11 000102  000003                           bpt
     12                                 ;
     13 000104  016702  000104                   mov     x,r2
     14 000110  000003                           bpt
     15                                 ;
     16 000112  012703  000123                   mov     #123,r3
     17 000116  000003                           bpt
     18                                 ;
     19 000120  000000                           halt
     20                                 ;
     21 000214                          . = base + 214
     22                                 ;
     23 000214  177777                  x:       .word   -1
     24                                 ;
     25 000216  000076'                          .end    start


base   = 000000 R  
pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000076 RG 
x        000214 R  
