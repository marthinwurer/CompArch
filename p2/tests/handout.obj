03e 2 03 0a
040 2 8c 10
042 2 03 00
044 4 c2 1d 44 00
048 2 03 00
04a 4 c3 15 53 00
04e 2 03 00
050 2 00 00
08c 2 ff ff
03e
