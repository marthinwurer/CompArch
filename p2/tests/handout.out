CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 76
000076:  PS[00]  005003  CLR R3
000100:  PS[04]  010214  MOV R2,(R4)
000102:  PS[04]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000104]
000104:  PS[04]  016702 000104  MOV 000104(R7),R2
000110:  PS[10]  000003  BPT
  R0[000000]  R1[000000]  R2[177777]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000112]
000112:  PS[10]  012703  MOV (R7)+,R3
000116:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[177777]  R3[000123]
  R4[000000]  R5[000000]  R6[000000]  R7[000120]
000120:  PS[00]  000000  HALT

Machine Halted - HALT instruction

Simulated time 47 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
